lib/file_transfer_agent/remote_storage_util.js

```diff
@@ -41,15 +43,17 @@ function RemoteStorageUtil(connectionConfig) {
   * @returns {Object}
   */
   this.getForStorageType = function (type) {
+    if (client) {
+      return client;
+    }
     if (type === 'S3') {
-      return new SnowflakeS3Util(connectionConfig);
+      client = new SnowflakeS3Util(connectionConfig);
     } else if (type === 'AZURE') {
-      return new SnowflakeAzureUtil(connectionConfig);
+      client = new SnowflakeAzureUtil(connectionConfig);
     } else if (type === 'GCS') {
-      return new SnowflakeGCSUtil();
-    } else {
-      return null;
+      client = new SnowflakeGCSUtil(connectionConfig);
     }
+    return client;
```

Comment:
when null - should we log sth?

Reply:
We have returned null if the type does not match any of the available Utils. As far as I understand, this implies that the driver will default to using local storage. Do you think adding a log for this scenario would be necessary?